pennylane/interfaces/jax_jit.py

```diff
@@ -91,174 +105,114 @@ def _jac_shape_dtype_struct(tape: "qml.tape.QuantumScript", device: "qml.Device"
-    finite differences grad transform).
+def _vjp_bwd(tapes, execute_fn, jpc, device, params, dy):
+    """Perform the backward pass of a vjp calculation, returning the vjp."""
 
-    Note: we are not using the batch_jvp pipeline and rather split the steps of unwrapping tapes and the JVP
-    computation because:
+    def wrapper(inner_params, inner_dy):
+        new_tapes = _set_parameters_on_copy(tapes.vals, inner_params)
+        return tuple(jpc.compute_vjp(new_tapes, inner_dy))
 
-    1. Tape unwrapping has to happen in the callback: otherwise jitting is broken and Tracer objects
-    are converted to NumPy, something that raises an error;
+    vjp_shape = _pytree_shape_dtype_struct(params)
+    return (jax.pure_callback(wrapper, vjp_shape, params, dy),)
 
-    2. Passing in the tangents as an argument to the wrapper function called by the jax.pure_callback raises an
-    error (as of jax and jaxlib 0.3.25):
-    ValueError: Pure callbacks do not support transpose. Please use jax.custom_vjp to use callbacks while
-    taking gradients.
 
-    Solution: Use the callback to compute the jacobian and then separately compute the JVP using the
-    tangent.
-    """
+_execute_jvp_jit = jax.custom_jvp(_execute_wrapper, nondiff_argnums=[1, 2, 3, 4])
+_execute_jvp_jit.defjvp(_execute_and_compute_jvp, symbolic_zeros=True)
 
-    def wrapper(inner_params):
-        new_tapes = set_parameters_on_copy_and_unwrap(tapes, inner_params, unwrap=False)
-        all_jacs = []
-        for new_t in new_tapes:
-            jvp_tapes, res_processing_fn = gradient_fn(new_t, **gradient_kwargs)
-            jacs = execute_fn(jvp_tapes)[0]
-            jacs = res_processing_fn(jacs)
-            all_jacs.append(jacs)
-        return tuple(all_jacs)
-
-    expected_shapes = tuple(_jac_shape_dtype_struct(t, device) for t in tapes)
-    return jax.pure_callback(wrapper, expected_shapes, params)
-
-
-def _grad_transform_jac_no_callback(
-    params, tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n, max_diff
-):
-    """Performs the gradient transform in a differentiable manner."""
-    new_tapes = set_parameters_on_copy_and_unwrap(tapes, params, unwrap=False)
-    jacobians = []
-    for new_t in new_tapes:
-        jvp_tapes, res_processing_fn = gradient_fn(new_t, **gradient_kwargs)
-        jacs = execute(
-            jvp_tapes,
-            device,
-            execute_fn,
-            gradient_fn,
-            gradient_kwargs,
-            _n=_n + 1,
-            max_diff=max_diff,
-        )
-        jacs = res_processing_fn(jacs)
-        jacobians.append(jacs)
-    return tuple(jacobians)
+_execute_vjp_jit = jax.custom_vjp(_execute_wrapper, nondiff_argnums=[1, 2, 3, 4])
+_execute_vjp_jit.defvjp(_vjp_fwd, _vjp_bwd)
 
 
-def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=1):
-    """Execute a batch of tapes with JAX parameters on a device.
+def jax_jvp_jit_execute(tapes, execute_fn, jpc, device):

```

Comment:
I would have expected the name to be `jax_jit_jvp_execute`, because both `jax` and `jit` are part of the interface name.